### ISSUE TITLE
Can't connect to Confluence

### ISSUE THREAD

**Issue by king-a** [tonybaloney/sphinxcontrib-confluencebuilder#2]
I can't connect to Confluence while publishing using sphinx-build with target confluence

here is the full traceback:
```

# Sphinx version: 1.4.8
# Python version: 3.5.2 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.8
# Last messages:
# Loaded extensions:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinxcontrib_confluencebuilder-0.2.0.dev20161117-py3.5.egg/sphinxcontrib/builders/confluence.py", line 201, in _connect
    password=self.config.confluence_server_pass)
  File "/usr/local/lib/python3.5/dist-packages/confluence-0.2.dev20161114154426-py3.5.egg/confluence/confluence.py", line 193, in __init__
    self._token = self._server.confluence1.login(username, password)
  File "/usr/lib/python3.5/xmlrpc/client.py", line 1092, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python3.5/xmlrpc/client.py", line 1432, in __request
    verbose=self.__verbose
  File "/usr/lib/python3.5/xmlrpc/client.py", line 1134, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib/python3.5/xmlrpc/client.py", line 1146, in single_request
    http_conn = self.send_request(host, handler, request_body, verbose)
  File "/usr/lib/python3.5/xmlrpc/client.py", line 1259, in send_request
    self.send_content(connection, request_body)
  File "/usr/lib/python3.5/xmlrpc/client.py", line 1289, in send_content
    connection.endheaders(request_body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.5/socket.py", line 693, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/lib/python3.5/socket.py", line 732, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/myUserName/.local/lib/python3.5/site-packages/sphinx/cmdline.py", line 243, in main
    opts.warningiserror, opts.tags, opts.verbosity, opts.jobs)
  File "/home/myUserName/.local/lib/python3.5/site-packages/sphinx/application.py", line 204, in __init__
    self._init_builder(self.buildername)
  File "/home/myUserName/.local/lib/python3.5/site-packages/sphinx/application.py", line 278, in _init_builder
    self.builder = builderclass(self)
  File "/home/myUserName/.local/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 87, in __init__
    self.init()
  File "/usr/local/lib/python3.5/dist-packages/sphinxcontrib_confluencebuilder-0.2.0.dev20161117-py3.5.egg/sphinxcontrib/builders/confluence.py", line 76, in init
    self._connect()
  File "/usr/local/lib/python3.5/dist-packages/sphinxcontrib_confluencebuilder-0.2.0.dev20161117-py3.5.egg/sphinxcontrib/builders/confluence.py", line 205, in _connect
    raise Exception("Could not connect, check remote API is configured. %s" % ex)
Exception: Could not connect, check remote API is configured. [Errno -2] Name or service not known


```

**Comment by tonybaloney** [comment]
hi @king-a did you set the `confluence_server_url` parameter? If not it shouldn't try and upload. If you did, it sounds like you put the wrong port in or it's not connecting because of the firewall

**Comment by king-a** [comment]
Hi @tonybaloney I indeed set `confluence_server_url` parameter. thanks for the help I 'll look in that direction.

**Comment by king-a** [comment]
Ok now I understand why i'm getting no connection, it's because I'm behind a proxy. on my first thought I've think Confluence was on an internal server, turn out it is actually outside so i'm experiencing proxy's issue. do you have any idea on how I can use confluence behind a proxy server ?

**Comment by tonybaloney** [comment]
found a potential solution to using a HTTP Proxy transport with urllib2 and xmlrpclib but it looks tricky. https://gist.github.com/nathforge/980961

**Comment by jdknight** [comment]
Proxy support for the XML-RPC API interaction has been introduced in the commit ced564581ab6640f5a29cc23b8b0e9a4acd606c6. Additional tests will be performed before merging into the `master`. Once merged, I believe we can close this issue.

**Comment by jdknight** [comment]
With recent development, proxy support should be available for both REST API and XML-RPC API. REST API, done through the Python `requests` module uses proxy information set in the environment (i.e. `http_proxy` and `https_proxy`). XML-RPC API can be configured to use a proxy via the `confluence_proxy` configuration value.

Support for proxy has been merged into `master` (see #20). This will be made available next release cycle (see #18). Marking as closed.
